src/include/cursor.i

```diff
@@ -201,16 +201,16 @@ __cursor_reset(WT_CURSOR_BTREE *cbt)
     if (F_ISSET(cbt, WT_CBT_ACTIVE)) {
         if (!F_ISSET(cbt, WT_CBT_NO_TRACKING))
             __cursor_leave(session);
+
+        /*
+         * When the count of active cursors in the session goes to zero, there are no active
+         * cursors, and we can release any snapshot we're holding for read committed isolation.
+         */
+        if (session->ncursors == 0 && !F_ISSET(cbt, WT_CBT_NO_TXN))
```

Comment:
@quchenhao I think the comment is misleading here because we don't check for the read committed isolation before calling `__wt_txn_release_snapshot`.

Reply:
Fixed. Thanks.